libs/ui/src/lib/input/numbers-only.directive.ts

```diff
@@ -0,0 +1,17 @@
+import { Directive, ElementRef, HostListener, Input } from '@angular/core';
+
+@Directive({
+  // eslint-disable-next-line @angular-eslint/directive-selector
+  selector: '[numbersOnly]',
+})
+export class NumbersOnlyDirective {
+  @Input() numbersOnly = false;
+  constructor(private el: ElementRef<HTMLInputElement>) {}
+
+  @HostListener('input', ['$event'])
+  onInput(event: InputEvent) {
+    if (this.numbersOnly) {
+      this.el.nativeElement.value = this.el.nativeElement.value.replace(/[^0-9]/g, '');
```

Comment:
This way to limit the character input range might not be useful for assistive devices because the limitation to numbers is not standard and therefore would not be communicated to them. 

It could make sense to look at an alternative implementation such as through using `inputmode="numeric"`. 

Some resources on the topic that i've found: https://www.magentaa11y.com/checklist-web/number-input/, https://www.coywolf.news/webdev/accessible-number-input/.

Reply:
Added inputmode="numeric" and pattern="[0-9]*", but still left the old logic as well. Only using pattern and inputmode will let users to copy text into that field as you can test in https://www.magentaa11y.com/checklist-web/number-input/ . Let me know if any further changes are required.